src/Filter.ts

```diff
@@ -58,41 +58,25 @@ export class ListFilter extends Filter {
 }
 
 export class CountryFilter extends ListFilter {
+    private groups: { [code: string]: string[] };
+
     constructor(settings: CountryFilterSettings) {
         super(settings);
-        this.options(settings.options);
+        this.groups = settings.groups;
+        this.options(settings.options || []);
         this.selectedOptions(settings.selected || settings.options);
     }
 
     public selectCountryGroup(selectedGroup: string) {
-        switch (selectedGroup) {
-            case "all":
-                this.selectedOptions([...this.options()]);
-                break;
-            case "none":
-                this.selectedOptions([]);
-                break;
-            case "pine":
-                this.selectedOptions([...pineCountries]);
-                break;
-            case "dove94":
-                this.selectedOptions([...dove94]);
-                break;
-            case "dove96":
-                this.selectedOptions([...dove96]);
-                break;
-            case "gavi77":
-                this.selectedOptions([...gavi77]);
-                break;
-            case "gavi72":
-                this.selectedOptions([...gavi72]);
-                break;
-            case "gavi68":
-                this.selectedOptions([...gavi68]);
-                break;
-            default:
-                this.selectedOptions([]);
-                break;
+        if (selectedGroup === "all") {
+            this.selectedOptions([...this.options()]);
+        } else if (selectedGroup === "none") {
+            this.selectedOptions([]);
+        } else if (this.groups[selectedGroup] !== undefined) {
+            this.selectedOptions(this.groups[selectedGroup]);
```

Comment:
Was there a reason that arrays were being cloned previously:
e.g. `this.selectedOptions([...pineCountries]);` as opposed to `this.selectedOptions(pineCountries);` and is this reason still applicable here? I'm wondering if the clone was because otherwise user actions that triggered updates to the selected options were modifying the original array.

Reply:
I have just checked and it looks like the cloning is not applicable any more.
I am not sure when the behaviour changed, but it is unnecessary now. I've updated line 72 to be consistent.